hack/netpol/README.md

```diff
@@ -0,0 +1,151 @@
+y_c
+--> map[x_a:false x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+z_a
+--> map[x_a:false x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+z_b
+--> map[x_a:true x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+z_c
+--> map[x_a:false x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+x_a
+--> map[x_a:true x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+x_b
+--> map[x_a:true x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+x_c
+--> map[x_a:false x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+y_b
+--> map[x_a:true x_b:true x_c:true y_a:true y_b:true y_c:true z_a:true z_b:true z_c:true]
+correct:81, incorrect:0, result=%!(EXTRA bool=true) true
+
+-	x/a	y/a	z/a	x/b	y/b	z/b	x/c	y/c	z/c
+x/a	.	.	.	.	.	.	.	.	.
+y/a	X	.	.	.	.	.	.	.	.
+z/a	X	.	.	.	.	.	.	.	.
+x/b	.	.	.	.	.	.	.	.	.
+y/b	.	.	.	.	.	.	.	.	.
+z/b	.	.	.	.	.	.	.	.	.
+x/c	X	.	.	.	.	.	.	.	.
+y/c	X	.	.	.	.	.	.	.	.
+z/c	X	.	.	.	.	.	.	.	.
+
+
+observed:
+
+-	x/a	y/a	z/a	x/b	y/b	z/b	x/c	y/c	z/c
+x/a	.	.	.	.	.	.	.	.	.
+y/a	X	.	.	.	.	.	.	.	.
+z/a	X	.	.	.	.	.	.	.	.
+x/b	.	.	.	.	.	.	.	.	.
+y/b	.	.	.	.	.	.	.	.	.
+z/b	.	.	.	.	.	.	.	.	.
+x/c	X	.	.	.	.	.	.	.	.
+y/c	X	.	.	.	.	.	.	.	.
+z/c	X	.	.	.	.	.	.	.	.
+
+
+comparison:
+
+-	x/a	y/a	z/a	x/b	y/b	z/b	x/c	y/c	z/c
+x/a	.	.	.	.	.	.	.	.	.
+y/a	.	.	.	.	.	.	.	.	.
+z/a	.	.	.	.	.	.	.	.	.
+x/b	.	.	.	.	.	.	.	.	.
+y/b	.	.	.	.	.	.	.	.	.
+z/b	.	.	.	.	.	.	.	.	.
+x/c	.	.	.	.	.	.	.	.	.
+y/c	.	.	.	.	.	.	.	.	.
+z/c	.	.	.	.	.	.	.	.	.
+
+```
+
+## How is this different then NetworkPolicy tests in upstream K8s ?
+
+We are working to merge this into upstream Kubernetes, in the meanwhile, here's the differences.
+
+- We define tests as *truth tables, and have a 'builder' library* for building up network policy structs with almost no boilerplate, meaning you can define a very sophisticated network policy test in just a few lines of code.
+- *Comprehensive:* All pod-to-pod connectivity is validated for every test run.  In a typical network policy test in current upstream we only validate 2 or 3 scenarios, leaving out intra and inner namespace connections which might be compromised due to a hard to detect CNI inconsistency.  In these tests, we test all 81 connections for 3 identical pods running in 3 different namespaces (i.e. the 9x9 connectivity matrix).
+- *Transparent:* Each test prints out a `kubectl` command you can run to re-probe a given pods connectivity patterns.
+- It's *fast:* Because we use `kubectl exec` to run tests with `wget` between pods, all 81 tests can easily finish in 20 seconds or less, even if pod scheduling is slow.  This is because no polling is done, and there is no down/uptime for pods.
+- *Easy to reason about:* The pods in this repo stay up forever, so you can reuse the above kubectl commands outputted by your netpol logs to exec into a pod and reproduce any failures.
```

Comment:
we probably also need a way to cleanup these pods as an option provided while running tests

Reply:
Im open to this, will ping you later to discuss.... 

first glance,  sounds like a bit of a follow on, given that this is meant mainly to be an API for building our internal tests out, as opposed to as an end-user facing thing.